auto-milestone/main.go

```diff
@@ -0,0 +1,230 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/google/go-github/v50/github"
+	"github.com/grafana/grafana-github-actions-go/pkg/ghgql"
+	"github.com/grafana/grafana-github-actions-go/pkg/toolkit"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+)
+
+func main() {
+	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr}).With().Timestamp().Logger()
+	ctx := context.Background()
+	ctx, cancel := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+	ctx = logger.WithContext(ctx)
+
+	var repo string
+	var doPreview bool
+	var listInputs bool
+	var prNumber int
+
+	pflag.StringVar(&repo, "repo", os.Getenv("GITHUB_REPOSITORY"), "owner/repo pair for a repository on GitHub")
+	pflag.BoolVar(&doPreview, "preview", false, "Only determine the milestone but don't set it")
+	pflag.BoolVar(&listInputs, "list-inputs", false, "Show a list of all available inputs")
+	pflag.Parse()
+
+	logger.Info().Msgf("Operating inside %s", repo)
+
+	rawPRNumber := pflag.Arg(0)
+	if rawPRNumber == "" {
+		logger.Fatal().Msg("No PR specified")
+		return
+	}
+	if parsed, err := strconv.ParseInt(rawPRNumber, 10, 32); err != nil {
+		logger.Fatal().Err(err).Msg("Failed to parse PR number")
+		return
+	} else {
+		prNumber = int(parsed)
+	}
+
+	// Determine the base-branch of that pull request
+	tk, err := toolkit.Init(
+		ctx,
+	)
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to initialize toolkit")
+	}
+
+	if listInputs {
+		tk.ShowInputList()
+		return
+	}
+	defer func() {
+		if err := tk.SubmitUsageMetrics(ctx); err != nil {
+			logger.Warn().Err(err).Msg("Failed to submit usage metrics")
+		}
+	}()
+
+	elems := strings.Split(repo, "/")
+	repoOwner := elems[0]
+	repoName := elems[1]
+
+	gh := tk.GitHubClient()
+
+	pr, _, err := gh.PullRequests.Get(ctx, repoOwner, repoName, prNumber)
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to fetch pull-request")
+	}
+	prTargetBranch := pr.GetBase()
+	if prTargetBranch == nil {
+		logger.Fatal().Msg("The requested pull-request has no target branch")
+		return
+	}
+
+	var targetMilestoneName string
+
+	content, _, _, err := gh.Repositories.GetContents(ctx, prTargetBranch.GetRepo().GetOwner().GetLogin(), prTargetBranch.GetRepo().GetName(), "package.json", &github.RepositoryContentGetOptions{Ref: prTargetBranch.GetRef()})
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to retrieve package.json of PR base")
+		return
+	}
+	if cnt, err := content.GetContent(); err != nil {
+		logger.Fatal().Err(err).Msg("Failed to retrieve package.json of PR base")
```

Comment:
Same error message with the one above, is that intentional?

Reply:
Mostly because I was lazy. Will be fixed soon. Thanks :)